### DISCUSSION TITLE
Python bindings [13653]

### DISCUSSION THREAD

**Discussion by maxikrie** [eProsima/Fast-DDS#2447] [Q&A]
Hello,

Are you planning to release a reference implementation for Python bindings or can you point me somewhere? I know that ROS2 has implemented bindings for Python, but they are one too many levels of abstraction for my taste and also employ the entire ROS2 ecosystem, which is not an option for me. I could use the ROS2 implementation as a starting point for my own implementation, however I don't want to duplicate work if you are planning to release this anyways.

In any case, I am very happy so far with Fast-DDS and Python bindings would even increase usability in my opinion.

Thanks!
Max

**Comment by JLBuenoLopez** [comment]
Hi @maxikrie,

Sorry for the late answer. There is some work being done in this line. You can find more information in the [Fast-DDS-Python repository](https://github.com/eProsima/Fast-DDS-python). It is still being developed and unstable, but we are working on releasing some documentation and complete the DDS API in the next few months!
